src/components/NavBar.js

```diff
@@ -0,0 +1,185 @@
+import logo from '../images/raptoreum-logo-ger.svg';
+import styled, { css } from 'styled-components';
+import { useState } from 'react';
+import { NavLink } from 'react-router-dom';
+
+export default function NavBar() {
+  const [click, setClick] = useState(false);
+
+  const handleClick = () => setClick(!click);
+  const closeMobileMenu = () => setClick(false);
+
+  return (
+    <NavBarHeader>
+      <Nav>
+        <LinkStyledLogo to="/" onClick={closeMobileMenu}>
+          <RaptoreumLogoContainer>
+            <RaptoreumLogoImage src={logo} alt="Raptoreum-Logo" />
+          </RaptoreumLogoContainer>
+          <RaptoreumDecryptdedContainer>
+            <RaptoreumDecryptedText>
+              raptoreum.de.crypted
+            </RaptoreumDecryptedText>
+          </RaptoreumDecryptdedContainer>
+        </LinkStyledLogo>
+        <MenuIconContainer onClick={handleClick}>
+          <MenuIcon
+            className={click ? 'fa-solid fa-times' : 'fa-solid fa-bars'}
```

Comment:
Maybe you can use React Icons here and rename the condition :) 
`className={showMenu ? 'fa-solid fa-times' : 'fa-solid fa-bars'}`